addons/website_form/controllers/main.py

```diff
@@ -32,6 +32,14 @@ def website_form(self, model_name, **kwargs):
             # I couldn't find a cleaner way to pass data to an exception
             return json.dumps({'error_fields' : e.args[0]})
 
+        geoip_country_code = request.session.geoip and request.session.geoip.get('country_code') or False
+        if geoip_country_code:
+            country_id = request.env['res.country'].sudo().search([('code', '=', geoip_country_code)], limit=1)
+            geoip_state_code = request.session.geoip and request.session.geoip.get('region') or False
+            if geoip_state_code:
+                state_id = request.env['res.country.state'].sudo().search([('code', '=', geoip_state_code), ('country_id', '=', country_id.id)], limit=1)
```

Comment:
idem that for country: ('name_code_uniq', 'unique(country_id, code)', 'The code of the state must be unique by country !')

Here, the limit=1 is more secure, because you will set it after on the record and that this constraint is not in the old versions, it is more secure to add it ! Some migration failed to add the constraint.